### ISSUE TITLE
Move recipes between userspaces

### ISSUE THREAD

**Issue by L1nKinc** [TandoorRecipes/recipes#1935]
### Is your feature request related to a problem? Please describe.

First of all the new userspace features are awesome and I'm looking forward to all the new possibilities.
I created a second userspace and wanted to move some of the existing recipes from the default space to the newly created one. The only option I found is to change the space assigned to the recipe itself in the admin panel. The result is, that the recipe is almost empty in the new userspace exept of the title and picture. So I found out, that I would have to also manually change the assigned space to all the steps the recipe consits of. That's very error prone and takes a lot of manual work.

### Describe the solution you'd like

Maybe you could add a feature in the recipe edit form to change the assigned space of the recipe, if the user has the rights in the old and the new space.

### Describe alternatives you've considered

Another option would be to make some sort of automated skript in the admin panel, so if you change the space of a recipe, all steps will also get moved to the new space.

**Comment by vabene1111** [comment]
nice idea, there are many interesting features that multi space tennancy will allow at some point

**Comment by j007bond007** [comment]
I agree, I have a use case for this utilizing spaces to share recipes, but yet each household have their own unique meal plan/grocery list and private recipes. It would really bring Tandoor to the next level of recipe management.

**Comment by vabene1111** [comment]
you can have individual meal plans, grocery lists and private recipes all within one space, still this is something that will likely be fixed/implemented at some point

**Comment by L1nKinc** [comment]
Any news on this?
Or is there a best practice workaround?
Thank a lot for your efforts and time!

**Comment by vabene1111** [comment]
currently there is not really a good way to do this and its pretty tricky to implement so do not expect this soon

**Comment by L1nKinc** [comment]
Thanks for your reply! I did not expect that it is so complicated. So I can understand that it's not on your priority list. Nevertheless, thanks for all your work!
If I have some free time maybe I will try to tinker with the database entries to find a "one time" solution / workaround to move existing recipes. If it works I can share my findings.

**Comment by vabene1111** [comment]
The main problem is the extrem nesting of data from recipe > steps > ingredients > foods > ... 

You need to create all those entries in the other space since they might not exist. The easiest way is likely to use the copy feature in the frontend and switch spaces between copies

**Comment by jakobw123** [comment]
Hey, first of all, thanks for this amazing tool. Really like it. The copying logic though would be a cool feature for me as well. From a design and implementation viewpoint, is copying different from the export/import logic you have already implemented for quite a few other data formats (other recipe apps, etc.)? Or how are the nested structures handled in export/import? Could the copying and export/import logic be merged? For merging I have in mind something like: Use the default export/import format, which can contain everything Tandoor can provide/allow. Then both copying and export/import logic would work on this like an interface data schema.

Maybe my use case is also helpful:
I use Tandoor as my recipe database, but also for planning events and trips. I want to reuse my private recipes in this event planning. I want different friends to be part of the planning for different events. So they should be allowed to change recipes freely, add new ones, etc.. But I do not want them or even me to change the original private recipe during the event planning. In that sense the recipe for the event shall be a true deepcopy of my private recipe. 

You stated above that, among other things, private recipes inside one single space are possible. Does it allow duplicate recipes? And if I would add all the different event plannings into the same single space, wouldn't it make the search for recipes (because of a lot of duplicates) and the separation of events really hard? Or is a separate space for different events actually the better approach to separate data and concerns? This use case also goes beyond recipes, I think. The separation issue I see there would also apply for grocery lists, meal plans, etc..

Feel free to critique. Just wanted to share my questions and thoughts on this.
